tests/ttnn/unit_tests/operations/test_all_gather.py

```diff
@@ -410,14 +410,15 @@ def run_line_all_gather(
     for i, t in enumerate(input_tensors):
         tt_input_tensors.append(ttl.tensor.Tensor(t, input_dtype).to(layout).to(devices[i], mem_config))
 
+    input_tensor_mesh = ttnn.aggregate_as_tensor(tt_input_tensors)
```

Comment:
Can we start moving away from the old-style way of dealing with these multi-device tensors?

For example:
```
    input_tensors = torch.chunk(input_tensor, num_devices, dim)
    tt_input_tensors = []
    for i, t in enumerate(input_tensors):
        tt_input_tensors.append(ttl.tensor.Tensor(t, input_dtype).to(layout).to(devices[i], mem_config))

    input_tensor_mesh = ttnn.aggregate_as_tensor(tt_input_tensors)
```
can be replaced with:
```
ttnn_tensor = ttnn.from_torch(full_tensor, mesh_mapper=ShardTensorToMesh(device_mesh, dim=dim))
```


Take a look at: `tests/ttnn/unit_tests/test_multi_device.py::test_ttnn_multi_device_all_gather`